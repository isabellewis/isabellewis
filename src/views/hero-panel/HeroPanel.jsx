import React, { PureComponent } from 'react';
import styled from 'styled-components';
import { TypingText } from '../../components';

const HeroWrapper = styled.div`
	animation: 2s fadeIn 1;
	background: #000;
	display: block;
	min-height: 100%;
	margin: 0;
	overflow: hidden;
	padding: 100px 30px;
	text-align: right;
	width: 100%;

	@keyframes fadeIn {
		0%, 30% {
			opacity: 0;
		}
		100% {
			opactity: 1;
		}
	}

	@media only screen and (min-width: 576px) {
		padding: 200px 50px;
	}

	@media only screen and (min-width: 992px) {
		padding: 300px 200px;
	}
`;

const headingText = ['Isabel', 'Iz', 'Izzy', 'Is', 'Izzy'];

export class HeroPanel extends PureComponent {
	constructor(props) {
		super(props);
	}

	render() {
		return (
			<HeroWrapper id="heroPanel">
				<TypingText textValues={headingText} />
			</HeroWrapper>
		);
	}
}