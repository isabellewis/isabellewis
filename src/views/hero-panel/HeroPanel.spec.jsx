import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { HeroPanel } from './HeroPanel';

describe('views/HeroPanel', () => {
	let component;

	beforeEach(() => {
		component = shallow(<HeroPanel />);
	});

	it('should match the snapshot', () => {
		expect(component).toMatchSnapshot();
	});
});