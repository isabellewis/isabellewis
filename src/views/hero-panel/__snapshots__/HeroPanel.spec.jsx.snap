// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/HeroPanel should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeroPanel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <TypingText
        textValues={
          Array [
            "Isabel",
            "Iz",
            "Izzy",
            "Is",
            "Izzy",
          ]
        }
      />,
      "id": "heroPanel",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "textValues": Array [
          "Isabel",
          "Iz",
          "Izzy",
          "Is",
          "Izzy",
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "HeroPanel__HeroWrapper-iDGzIU",
        "isStatic": true,
        "rules": Array [
          "animation:2s fadeIn 1;background:#000;display:block;min-height:100%;margin:0;overflow:hidden;padding:100px 30px;text-align:right;width:100%;@keyframes fadeIn{0%,30%{opacity:0;}100%{opactity:1;}}@media only screen and (min-width:576px){padding:200px 50px;}@media only screen and (min-width:992px){padding:300px 200px;}",
        ],
      },
      "displayName": "HeroPanel__HeroWrapper",
      "render": [Function],
      "styledComponentId": "HeroPanel__HeroWrapper-iDGzIU",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <TypingText
          textValues={
            Array [
              "Isabel",
              "Iz",
              "Izzy",
              "Is",
              "Izzy",
            ]
          }
        />,
        "id": "heroPanel",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "textValues": Array [
            "Isabel",
            "Iz",
            "Izzy",
            "Is",
            "Izzy",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "HeroPanel__HeroWrapper-iDGzIU",
          "isStatic": true,
          "rules": Array [
            "animation:2s fadeIn 1;background:#000;display:block;min-height:100%;margin:0;overflow:hidden;padding:100px 30px;text-align:right;width:100%;@keyframes fadeIn{0%,30%{opacity:0;}100%{opactity:1;}}@media only screen and (min-width:576px){padding:200px 50px;}@media only screen and (min-width:992px){padding:300px 200px;}",
          ],
        },
        "displayName": "HeroPanel__HeroWrapper",
        "render": [Function],
        "styledComponentId": "HeroPanel__HeroWrapper-iDGzIU",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
